src/Compiler/lex.fsl

```diff
@@ -530,11 +530,13 @@ rule token (args: LexArgs) (skip: bool) = parse
  | '\'' trigraph '\'' 'B'
      { let s = lexeme lexbuf
        let x = int32 (trigraph s.[2] s.[3] s.[4])
-       if x < 0 || x > 255 then
+       if x < 0 || x > 127 then
```

Comment:
this would be a breaking change, right? Code which was compiling before (and overflowing, likely), will be producing the warning now? As would one below (which adds check for the range).

Reply:
> warning 

In fact even compiler error (same as already existing checks for other notations do)


But yes. Both are now errors which previously compiled:
* `'\250'B`: Prev: ok, byte `250y`. Now: compiler error
  * Note: it didn't overflow because `250` is still inside byte range
* `"\937"B`: prev: Ok, byte array `[|169y|]`. Now: compiler error

Reply:
In this case, we probably shouldn't produce an error, but give a warning (and want people that it's giving bad results at runtime, and will be an error in future) which will be promoted to the error in future version, since we can't just break people's projects (though it's giving bad results at runtime).

Reply:
Trigraph ASCII Byte  
(Note: I incorporated https://github.com/dotnet/fsharp/issues/15868#issuecomment-1706684448 )

```fsharp
// prev
> '\973'B;;
  ^^^^^^^
stdin(11,1): error FS1157: This is not a valid byte literal

> '\250'B;;
val it: byte = 250uy


// now
> '\973'B;;
  ^^^^^^^
stdin(2,1): error FS1157: This is not a valid ASCII byte literal. Value must be < 128y.

> '\250'B;;
  ^^^^^^^
stdin(3,1): warning FS1157: This is not a valid ASCII byte literal. Value should be < 128y.
Note: In a future F# version this Warning will be promoted to Error!

val it: byte = 250uy
```

Note: I reuse the same error code for the warning. Considering the warning should get merged into the error it makes sense.  
But I don't know if this might have any consequences in build tools.

<br/>
<br/>


Trigraph String
```fsharp
// prev
> "\937";;
val it: string = "©"


// now
> "\937";;
  -^^^^
stdin(4,2): warning FS1252: '\937' is not a valid character literal.
Note: Currently the value is wrapped around byte range to '\169'. In a future F# version this Warning will be promoted to Error!

val it: string = "©"
```


<br/>
<br/>
<br/>

(not yet pushed to GitHub because I haven't adjusted the tests yet)